python/cugraph-equivariant/cugraph_equivariant/tests/test_tensor_product_conv.py

```diff
@@ -0,0 +1,115 @@
+        [(7,), nn.GELU(), (2, 3, 2)],
+        [None, None, None],
+    ],
+)
+def test_tensor_product_conv_equivariance(
+    mlp_channels, mlp_activation, scalar_sizes, batch_norm, e3nn_compat_mode
+):
+    torch.manual_seed(12345)
+
+    in_irreps = o3.Irreps("10x0e + 10x1e")
+    out_irreps = o3.Irreps("20x0e + 10x1e")
+    sh_irreps = o3.Irreps.spherical_harmonics(lmax=2)
+
+    tp_conv = FullyConnectedTensorProductConv(
+        in_irreps=in_irreps,
+        sh_irreps=sh_irreps,
+        out_irreps=out_irreps,
+        mlp_channels=mlp_channels,
+        mlp_activation=mlp_activation,
+        batch_norm=batch_norm,
+        e3nn_compat_mode=e3nn_compat_mode,
+    ).to(device)
+
+    num_src_nodes, num_dst_nodes = 9, 7
+    num_edges = 40
+    src = torch.randint(num_src_nodes, (num_edges,), device=device)
+    dst = torch.randint(num_dst_nodes, (num_edges,), device=device)
+    edge_index = torch.vstack((src, dst))
+
+    src_pos = torch.randn(num_src_nodes, 3, device=device)
+    dst_pos = torch.randn(num_dst_nodes, 3, device=device)
+    edge_vec = dst_pos[dst] - src_pos[src]
+    edge_sh = o3.spherical_harmonics(
+        tp_conv.sh_irreps, edge_vec, normalize=True, normalization="component"
+    ).to(device)
+    src_features = torch.randn(num_src_nodes, in_irreps.dim, device=device)
+
+    rot = o3.rand_matrix()
+    D_in = tp_conv.in_irreps.D_from_matrix(rot).to(device)
+    D_sh = tp_conv.sh_irreps.D_from_matrix(rot).to(device)
+    D_out = tp_conv.out_irreps.D_from_matrix(rot).to(device)
+
+    if mlp_channels is None:
+        edge_emb = torch.randn(num_edges, tp_conv.tp.weight_numel, device=device)
+        src_scalars = dst_scalars = None
+    else:
+        if scalar_sizes:
+            edge_emb = torch.randn(num_edges, scalar_sizes[0], device=device)
+            src_scalars = (
+                None
+                if scalar_sizes[1] == 0
+                else torch.randn(num_src_nodes, scalar_sizes[1], device=device)
+            )
+            dst_scalars = (
+                None
+                if scalar_sizes[2] == 0
+                else torch.randn(num_dst_nodes, scalar_sizes[2], device=device)
+            )
+        else:
+            edge_emb = torch.randn(num_edges, tp_conv.mlp[0].in_features, device=device)
+            src_scalars = dst_scalars = None
+
+    # rotate before
+    out_before = tp_conv(
+        src_features=src_features @ D_in,
+        edge_sh=edge_sh @ D_sh,
+        edge_emb=edge_emb,
+        graph=(edge_index, (num_src_nodes, num_dst_nodes)),
+        src_scalars=src_scalars,
+        dst_scalars=dst_scalars,
+    )
+
+    # rotate after
+    out_after = (
+        tp_conv(
+            src_features=src_features,
+            edge_sh=edge_sh,
+            edge_emb=edge_emb,
+            graph=(edge_index, (num_src_nodes, num_dst_nodes)),
+            src_scalars=src_scalars,
+            dst_scalars=dst_scalars,
+        )
+        @ D_out
+    )
```

Comment:
```suggestion
    out_before = tp_conv(
        src_features=src_features @ D_in.T,
        edge_sh=edge_sh @ D_sh.T,
        edge_emb=edge_emb,
        graph=(edge_index, (num_src_nodes, num_dst_nodes)),
        src_scalars=src_scalars,
        dst_scalars=dst_scalars,
    )

    # rotate after
    out_after = (
        tp_conv(
            src_features=src_features,
            edge_sh=edge_sh,
            edge_emb=edge_emb,
            graph=(edge_index, (num_src_nodes, num_dst_nodes)),
            src_scalars=src_scalars,
            dst_scalars=dst_scalars,
        )
        @ D_out.T
    )
```

Without the transpose we are relying on the fact that these representations are orthonormal and that the inverse rotation is in the group of rotations. With the `.T` it's closer to the definition of equivariance.